libs/traits/src/liquidity_pools.rs

```diff
@@ -34,11 +35,26 @@ pub trait LPEncoding: Sized {
 	/// It's the identity message for composing messages with pack_with
 	fn empty() -> Self;
 
-	/// Retrieves the message proof hash, if the message is a proof type.
-	fn get_proof(&self) -> Option<Proof>;
+	/// Returns whether the message is a proof or not.
+	fn is_proof_message(&self) -> bool;
```

Comment:
@wischli @lemunozm I added this and changed `get_message_hash` below.

Reply:
I'm not fully sure about adding a new method... 🤔 

I do not want to impose my model, but why not this: https://github.com/centrifuge/centrifuge-chain/pull/1961#discussion_r1718314551, is there anything that does not match with the expected behavior?

Reply:
Not a big fan of the new method but we use it here now:
- https://github.com/centrifuge/centrifuge-chain/blob/feat/lp-v2-message-recovery/pallets/liquidity-pools-gateway/src/message_processing.rs#L466
- https://github.com/centrifuge/centrifuge-chain/blob/feat/lp-v2-message-recovery/pallets/liquidity-pools-gateway/src/message_processing.rs#L98

Reply:
Ah ok 👍🏻 

Maybe my proposal doesn't fit because you need to forcibly check if the proof is proof. So you need 3 things:

- a `MessageHash`
- Create a `MessageProof`
- Check if the current message is a `MessageProof`

Yeah, let's go with this new method